lib/kvdb/kvdb_rest.c

```diff
@@ -1214,6 +1214,7 @@ kvs_query_tree(
             bad |= !cJSON_AddNumberToObject(kvset, "vblocks", metrics.num_vblocks);
             bad |= !cJSON_AddNumberToObject(kvset, "vgroups", metrics.vgroups);
             bad |= !cJSON_AddStringToObject(kvset, "rule", cn_rule2str(metrics.rule));
+            bad |= !cJSON_AddStringToObject(kvset, "misc", view->ekbuf);
```

Comment:
lazy name choice :)

Reply:
What are you storing in ekbuf?

Reply:
dude.  misc stuff.  should be obv. :)

Reply:
rofl

Reply:
Yep, couldn't think of anything better, suggestions?

Reply:
I did change the name of ekbuf to just "buf", but then I reverted it, too many unnecessary changes.  I could rename it if y'all want me to...

Reply:
Depends on what the content is of the buffer

Reply:
my point is when i see a field named "misc" in the REST output, i will have no idea what it is.  so do tell, what does this new field contain?

Reply:
I see.  I'll change it to "status".  If the kvset is part of a compaction job then it prints out the action, rule, time, and wait-message for that job.

Reply:
Oh, wait, this field only exists for 'k' rows, so "status" should do it...

Reply:
Done.

Reply:
> action, rule, time, and wait-message

At least some of these already exist in the response. Why the duplication?

Reply:
That seems very unlikely, can you point it out to me?  The status is info about what's happening right now, which is different than the current frozen state.

Reply:
`rule` is already in the response on a per kvset basis. `action` seems to not currently exist, so add that as a separate field in the response. Add separate fields for time and wmesg as well

Reply:
I'm not being clear.  The new info is data from a currently running compaction operation which indicates what it's currently doing, how long it's taking, and where it's sleeping.  The rule for the kvset is the rule from the compaction operation that created that kvset, while the action/rule in the status line is the rule being used to create a new kvset.
Separate fields are unnecessary, the status is ephemeral and present for only a few kvsets at random times and usually not for longer than 30s or so, whereas all the rest of the state is persistent for the the life of the kvset as kvsets are read-only.

Reply:
Why can't structured data have a "notes" field with free-form content?  With free-form content I can easily change it as needed and not have to worry about breaking some user that is expecting things never to change and/or get approval from a committee.  If I make it all structured then I end up with a whole bunch of fields that are mostly empty most and then have to support it til the end of time.  This also just clutters up the tabular output with a bunch of mostly empty fields that otherwise nicely overlap with the space used by the node rows for the edge keys.
Backing up, I use this stuff extensively, and I need it to produce the information I need to see in order to understand what's happening to the tree in order to fix or improve things.  The info that I need to see changes over time, so I need to ensure that the effort to change/update it is fairly low, I grow weary of fighting for every little thing...

Reply:
> This also just clutters up the tabular output with a bunch of mostly empty fields that otherwise nicely overlap with the space used by the node rows for the edge keys.

This indicates to me that the tabular output is not the right way to visualize the data. Even then, the whole point of the custom view for the tabular output of this particular is that you can take the REST response and mold it into what you want by editing the kvs-cn-tree-get.c file in hsettp. In addition, the tabular outputs support only showing columns that you care about.

Point being if you want this one column of jumbled data, do it in kvs-cn-tree-get.c and not in the REST responses that I worked so hard to clean up.

> If I make it all structured then I end up with a whole bunch of fields that are mostly empty most and then have to support it til the end of time.

Not true. The REST APIs are not set in stone, and I would advocate for some of the REST routes to only exist for developer builds.

Reply:
We don't have the resources to build better visualization tools.  So we must compromise.  Using a tabular view for rows of different types (kvsets, node, tree) is an example of compromise.  If that means we have to use an unstructured field here and there, I'm fine with it.  I'm a fan of normalized schemas (i.e., not embedding structure into values) but keeping things 100% normalized has cost.